### PR TITLE
Add cacheonly option to yum and dnf modules

### LINKED ISSUE #69397: Feature: yum --cacheonly
##### SUMMARY

Add a --cacheonly option to the ansible yum module, similar to the --downloadonly option.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

lib/ansible/modules/packaging/os/yum.py

##### ADDITIONAL INFORMATION

We have a group of Oracle servers that have no repository management server.  They update directly from Oracle's repositories.  Our thought is to do a "yum update --downloadonly" on all dev and prod servers.  Then we can update the dev servers first using "yum update --cacheonly" to test then updated and run the same command on the prod at a later time to make them consistent with the dev.

```yaml

```

### PR DESCRIPTION
**Pull request by aminvakil**
##### SUMMARY
Fixes https://github.com/ansible/ansible/issues/69397

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lib/ansible/module_utils/yumdnf.py
lib/ansible/modules/yum.py
lib/ansible/modules/dnf.py

##### ADDITIONAL INFORMATION
This has been previously opened in https://github.com/ansible/ansible/pull/70912.

### CODE REVIEW

**Comment by aminvakil** [comment]
ready_for_review

File: lib/ansible/modules/dnf.py [hunk 1]
```diff
@@ -228,6 +228,12 @@
     type: bool
     default: "no"
     version_added: "2.11"
+  cacheonly:
+    description:
+      - Tells dnf to run entirely from system cache; does not download or update metadata.
+    type: bool
+    default: "no"
+    version_added: "2.11"
```
Comment (mattclay) [inline lib/ansible/modules/dnf.py]:
```suggestion
    version_added: "2.12"
```

File: lib/ansible/modules/yum.py [hunk 2]
```diff
@@ -236,6 +236,12 @@
     version_added: "1.5"
     default: "yes"
     type: bool
+  cacheonly:
+    description:
+      - Tells yum to run entirely from system cache; does not download or update metadata.
+    default: "no"
+    type: bool
+    version_added: "2.11"
```
Comment (mattclay) [inline lib/ansible/modules/yum.py]:
```suggestion
    version_added: "2.12"
```

**Review by mattclay — COMMENTED** [review]
We're past the 2.11 cutoff for features, so this should be updated for 2.12. However, tests won't pass until we actually create the stable-2.11 branch.

**Comment by s-hertel** [comment]
@aminvakil Thanks for the contribution!

**Pull request merged**